apps/explorer/src/components/Activity/components/TransactionsActivityTable.tsx

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { ArrowRight12 } from '@mysten/icons';
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+
+import { genTableDataFromTxData } from '../../transactions/TxCardUtils';
+
+import { useGetTransactionBlocks } from '~/hooks/useGetTransactionBlocks';
+import { Link } from '~/ui/Link';
+import { Pagination } from '~/ui/Pagination';
+import { PlaceholderTable } from '~/ui/PlaceholderTable';
+import { TableCard } from '~/ui/TableCard';
+import { Text } from '~/ui/Text';
+import { numberSuffix } from '~/utils/numberUtil';
+
+const DEFAULT_TRANSACTIONS_LIMIT = 20;
+
+interface Props {
+    disablePagination?: boolean;
+    refetchInterval?: number;
+    initialLimit?: number;
+}
+
+export function TransactionsActivityTable({
+    disablePagination,
+    initialLimit = DEFAULT_TRANSACTIONS_LIMIT,
+}: Props) {
+    const [currentPage, setCurrentPage] = useState(0);
+    const [limit, setLimit] = useState(initialLimit);
+    const rpc = useRpcClient();
+    const {
+        data,
+        isLoading,
+        isFetching,
+        isFetchingNextPage,
+        fetchNextPage,
+        hasNextPage,
+    } = useGetTransactionBlocks(undefined, limit);
+
+    const countQuery = useQuery(
+        ['transactions', 'count'],
+        () => rpc.getTotalTransactionBlocks(),
+        { cacheTime: 24 * 60 * 60 * 1000, staleTime: Infinity, retry: false }
+    );
+
+    const cardData =
+        data && Boolean(data.pages[currentPage])
+            ? genTableDataFromTxData(data.pages[currentPage].data)
+            : undefined;
+
+    const isPaginated =
```

Comment:
Why isn't this just `!disablePagiantion`? I don't think these other conditions really matter